liferay-search/liferay-portal
# SXPElement Translations ;-;

### **ticket**: https://liferay.atlassian.net/browse/LPS-198941


### **Code Notes**

- Need to create upgrade sxpElements and sxpBlueprints

### **Use Cases to Test**

❌ Not working yet
✅ Working with this code
🧪 Need to test yet
❓ Need to be defined

- Portal Startup with en-US not available in available language ✅ 
```
company.default.locale=pt_BR
locales=pt_BR, en_GB, es_ES
locales.enabled=pt_BR, en_GB, es_ES
user.country=BR
user.language=pt
```

- Import an SXPElement JSON with title_i18n without the current locale. ✅ 
- Import an SXPElement JSON with title_i18n without any title for the available languages. ✅ 
- Copy an SXPElement lacking the default locale in the title_i18n map and ensure the title remains populated. ✅ 
- Add an SXPElement into an SXPBlueprint and save. ✅ 
- Add an SXPElement to an SXPBlueprint, save, export, then import. Ensure no title appears blank. ✅ 
- Create an SXPElement using en_US, then navigate to instance settings and disable en_US. Confirm all SXPElements continue to display titles. ✅ 
- After the Upgrade, all SXPElement have the two new fields  ✅ 
- After the Upgrade, all SXPElement inside the SXPBlueprints have the two new fields  ✅ 
- SXPElement Preview doesn't have a blank title when the default language is not available  ✅
